docs/features/mcp/README.md

```diff
@@ -0,0 +1,324 @@
+Remote WordPress MCP Server (mcp-adapter)
+    ↓ (WordPress Abilities API)
+WooCommerce Abilities
+    ↓ (REST API calls or direct operations)
+WooCommerce Core
+```
+
+### Architecture Components
+
+**Local MCP Proxy** (`mcp-wordpress-remote`)
+
+- Runs locally on the developer's machine as a Node.js process
+- Converts MCP protocol messages to HTTP requests
+- Handles authentication header injection
+- Bridges the protocol gap between MCP clients and WordPress REST endpoints
+
+**Remote WordPress MCP Server** (`mcp-adapter`)
+
+- Runs within WordPress as a plugin
+- Exposes the `/wp-json/woocommerce/mcp` endpoint
+- Processes incoming HTTP requests and converts them to MCP protocol messages
+- Manages tool discovery and execution
+
+#### WordPress Abilities System
+
+- Provides a standardized way to register and execute capabilities
+- Acts as an abstraction layer between MCP tools and actual operations
+- Enables flexible implementation approaches (REST bridging, direct DB operations, etc.)
+
+### Core Components
+
+**MCP Adapter Provider** ([`MCPAdapterProvider.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/MCP/MCPAdapterProvider.php))
+
+- Manages MCP server initialization and configuration
+- Handles feature flag checking (`mcp_integration`)
+- Provides ability filtering and namespace management
+
+**Abilities Registry** ([`AbilitiesRegistry.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Abilities/AbilitiesRegistry.php))
+
+- Centralizes WooCommerce ability registration
+- Bridges WordPress Abilities API with WooCommerce operations
+- Enables ability discovery for the MCP server
+
+**REST Bridge Implementation** ([`AbilitiesRestBridge.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Abilities/AbilitiesRestBridge.php))
+
+- Current preview implementation that maps REST operations to WordPress abilities
+- Provides explicit ability definitions with schemas for products and orders
+- Demonstrates how abilities can be implemented using existing REST controllers
+
+**WooCommerce Transport** ([`WooCommerceRestTransport.php`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/MCP/Transport/WooCommerceRestTransport.php))
+
+- Handles WooCommerce API key authentication
+- Enforces HTTPS requirements
+- Validates permissions based on API key scope
+
+### Implementation Approach
+
+For this developer preview, WooCommerce abilities are implemented by bridging to existing REST API endpoints. This approach allows us to quickly expose core functionality while leveraging proven REST controllers. However, the WordPress Abilities API is designed to be flexible - abilities can be implemented in various ways beyond REST endpoint proxying, including direct database operations, custom business logic, or integration with external services.
+
+## Enabling MCP Integration
```

Comment:
If it's possible to enable it via WC CLI, we should call it out here as an option. I believe this would be the command:

```
wp option update woocommerce_feature_mcp_integration_enabled yes
```